File: Sources/CountriesFlagsObjC/source/Countries/UIImage+BurkinaFaso.m
//
//  UIImage+BurkinaFaso.m
//  CountriesFlags
//

#import "Countries/UIImage+BurkinaFaso.h"

#import "Extensions/UIBezierPath+Star.h"
#import "Extensions/CGSize+Extension.h"

@implementation UIImage (BurkinaFaso)

+ (UIImage * _Nullable)flagBurkinaFasoWithSize:(CGSize)size {
    UIColor* colorRed = [UIColor colorWithRed:239.0f/255.0f
                                        green:43.0f/255.0f
                                         blue:45.0f/255.0f
                                        alpha:1.0f];
    UIColor* colorYellow = [UIColor colorWithRed:252.0f/255.0f
                                           green:209.0f/255.0f
                                            blue:22.0f/255.0f
                                           alpha:1.0f];
    UIColor* colorGreen = [UIColor colorWithRed:0.0f
                                          green:158.0f/255.0f
                                           blue:73.0f/255.0f
                                          alpha:1.0f];
    
    UIGraphicsBeginImageContextWithOptions(size, YES, 0.0f);
    
    [colorRed setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorGreen setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 2.0f, size.width, size.height / 2.0f));
    
    [colorYellow setFill];
    CGFloat radius = (size.height * 7.0f / 20.0f) / 2.0f;
    [[UIBezierPath starAtPoint:CGSizeMidPoint(size)
                        radius:radius
                          type:FlagStarTypeFivePointed] fill];
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+BurkinaFaso.swift
//
//  UIImage+BurkinaFaso.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Burkina_Faso "wikipedia"
     
     Generates the flag of Burkina Faso.
     For more information see [wikipedia].
     
     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Burkina Faso.
     */
    internal static func flagBurkinaFaso(with size: CGSize) -> UIImage? {
        let colorRed = UIColor(red: 239.0/255.0,
                               green: 43.0/255.0,
                               blue: 45.0/255.0,
                               alpha: 1.0)
        let colorYellow = UIColor(red: 252.0/255.0,
                                  green: 209.0/255.0,
                                  blue: 22.0/255.0,
                                  alpha: 1.0)
        let colorGreen = UIColor(red: 0.0,
                                 green: 158.0/255.0,
                                 blue: 73.0/255.0,
                                 alpha: 1.0)
        
        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)
        
        colorRed.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorGreen.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 2.0),
                          size: CGSize(width: size.width, height: size.height / 2.0)))
        
        colorYellow.setFill()
        let radius = CGFloat(size.height * 7.0 / 20.0) / 2.0
        UIBezierPath.star(with: size, radius: radius).fill()
        
        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()
        
        return image
    }
}

#endif
